tools/extra/fpgadiag/eth_group.cpp

```diff
@@ -54,140 +54,49 @@
 
 
 // open eth group
-int eth_group::eth_group_open(int vfio_id, std::string fpga_mdev_str)
+int eth_group::eth_group_open(std::string fpga_uid_str)
 {
-	char dev_path[256] = { 0 };
-	int i = 0;
-	struct vfio_group_status group_status ;
-	struct vfio_iommu_type1_info iommu_info;
-	struct vfio_iommu_type1_dma_map dma_map;
+	int res       = 0;
+	uint8_t *mem  = NULL;
 
-	memset(&group_status, 0, sizeof(group_status));
-	group_status.argsz = sizeof(group_status);
-
-	memset(&iommu_info, 0, sizeof(iommu_info));
-	iommu_info.argsz = sizeof(iommu_info);
-
-	memset(&dma_map, 0, sizeof(dma_map));
-	dma_map.argsz = sizeof(dma_map);
-
-	memset(&device_info, 0, sizeof(device_info));
-	device_info.argsz = sizeof(device_info);
-
-	memset(&device_info, 0, sizeof(device_info));
-	device_info.argsz = sizeof(device_info);
-
-	container = open(DEV_VFIO_PATH, O_RDWR);
-	if (container < 0) {
-		fprintf(stderr, "error opening container: %s\n",
-			strerror(errno));
-		return -1;
+	res = opae_uio_open(&uio, fpga_uid_str.c_str());
+	if (res) {
+		return res;
 	}
 
-	if (ioctl(container, VFIO_GET_API_VERSION) != VFIO_API_VERSION) {
-		printf("Wrong VFIO_API_VERSION \n");
-		return -1;
-	}
-	
-	if (!ioctl(container, VFIO_CHECK_EXTENSION, VFIO_TYPE1_IOMMU)) {
-		printf("Doen't support VFIO_TYPE1_IOMMU \n");
-		return -1;
+	res = opae_uio_region_get(&uio, 0, (uint8_t **)&mem, NULL);
+	if (res) {
+		return res;
 	}
 
-	if (snprintf(dev_path, sizeof(dev_path),
-		"%s/%d", "/dev/vfio", vfio_id) < 0) {
-		printf("snprint fails \n");
-		return -1;
-	}
-
-	group = open(dev_path, O_RDWR);
-	if (group < 0) {
-		fprintf(stderr, "error opening group: %s\n",
-			strerror(errno));
-		return -1;
-	}
-
-	// checks VFIO group status
-	ioctl(group, VFIO_GROUP_GET_STATUS, &group_status);
-	if (!(group_status.flags & VFIO_GROUP_FLAGS_VIABLE)) {
-		printf("Wrong VFIO_GROUP_FLAGS_VIABLE \n");
-		return -1;
-	}
 
-	// Add the group to the container
-	ioctl(group, VFIO_GROUP_SET_CONTAINER, &container);
+	mmap_ptr = mem;
+	ptr_ = (uint64_t*)mem;
 
-	// Enable the IOMMU model
-	ioctl(container, VFIO_SET_IOMMU, VFIO_TYPE1_IOMMU);
+	eth_info.csr = *(ptr_ + 1);
+	eth_dfh.csr = *(ptr_);
```

Comment:
Is there any identifying information in the DFH or the eth_info (like a GUID) that can be used to verifythe registers before poking them?

Reply:
Check with Feature ID, doesn't support GUID.
	// Check ETH group FeatureID
	if (eth_dfh.id != ETH_GROUP_FEATUREID) {
		printf("Worng Eth group Feature ID \n");
		return -1;
	}